addon/components/o-s-s/smart/tag-input.ts

```diff
@@ -0,0 +1,95 @@
+import { action } from '@ember/object';
+import { isBlank } from '@ember/utils';
+import { inject as service } from '@ember/service';
+import type { IntlService } from 'ember-intl';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import type { TagType } from './tag';
+
+export type Keyword = {
+  value: string;
+  type: TagType;
+};
+
+interface OSSSmartTagInputArgs {
+  value: string;
+  onKeydown(keyword: Keyword): void;
+  loading?: boolean;
+  hasError?: boolean;
+}
+
+export default class OSSSmartTagInput extends Component<OSSSmartTagInputArgs> {
+  @service declare intl: IntlService;
+
+  @tracked inputValue: string = this.args.value || '';
+  @tracked isInputFocused: boolean = false;
+  @tracked declare element: HTMLElement;
+
+  get keywordInputClasses(): string {
+    const classes = ['tag-input'];
+    if (this.isInputFocused) {
+      classes.push('tag-input--focus');
+    }
+    if (isBlank(this.inputValue)) {
+      classes.push('tag-input--empty');
+    }
+    if (this.args.hasError) {
+      console.log('input has error');
+      classes.push('tag-input--error');
+    }
+    return classes.join(' ');
+  }
+
+  get placeholder(): string {
+    return this.intl.t('oss-components.smart.tag_input.placeholder');
```

Comment:
I think we can also add an argument for this in case we need to overload the placeholder 🙏